### ISSUE TITLE
R2R images are not properly loaded when using single-file publish on Windows

### ISSUE THREAD

**Issue by agocke** [dotnet/runtime#39409]
Repro:

Publish a hello-world app on Windows as single-file

```
dotnet publish -r win-x64 -c release --self-contained true -p:PublishReadyToRun=true -p:PublishSingleFile=true net5.csproj
```

Now test the time (I did this using Git Bash):

```
$ time for i in {1..10}; do ./bin/release/net5/win-x64/publish/net5.exe > /dev/null; done

real    0m1.040s
user    0m0.076s
sys     0m0.106s
```

Now check the same thing vs the non-single-file publish:

```
$ time for i in {1..10}; do ./bin/release/net5/win-x64/net5.exe > /dev/null; done

real    0m0.747s
user    0m0.061s
sys     0m0.090s
```

If you disable R2R images on the non-single-file version, you get numbers close to the single-file:

```
$ time for i in {1..10}; do COMPlus_ReadyToRun=0 ./bin/release/net5/win-x64/net5.exe > /dev/null; done

real    0m1.096s
user    0m0.061s
sys     0m0.168s
```

and the `COMPlus_ReadyToRunLogFile=log.txt` file for single file R2R does seem to contain problems:

```
Ready to Run disabled - module not loaded for execution: "System.Private.CoreLib.dll".
Ready to Run disabled - module not loaded for execution: "C:\Users\andy\tmp\net5\bin\release\net5\win-x64\publish\net5.dll".
Ready to Run disabled - module not loaded for execution: "C:\Users\andy\tmp\net5\bin\release\net5\win-x64\publish\System.Runtime.dll".
Ready to Run disabled - module not loaded for execution: "C:\Users\andy\tmp\net5\bin\release\net5\win-x64\publish\System.Console.dll".
Ready to Run disabled - module not loaded for execution: "C:\Users\andy\tmp\net5\bin\release\net5\win-x64\publish\System.Threading.dll".
Ready to Run disabled - module not loaded for execution: "C:\Users\andy\tmp\net5\bin\release\net5\win-x64\publish\System.Text.Encoding.Extensions.dll".
Ready to Run disabled - module not loaded for execution: "C:\Users\andy\tmp\net5\bin\release\net5\win-x64\publish\System.Runtime.InteropServices.dll".
```

**Comment by agocke** [comment]
@VSadov Could you look at this as next? We'll need to fix this for 5.0

**Comment by VSadov** [comment]
Fixed in https://github.com/dotnet/runtime/pull/40104
